app/include/user_config.h

```diff
@@ -1,37 +1,206 @@
 #ifndef __USER_CONFIG_H__
 #define __USER_CONFIG_H__
 
-// #define FLASH_512K
-// #define FLASH_1M
-// #define FLASH_2M
-// #define FLASH_4M
-// #define FLASH_8M
-// #define FLASH_16M
+// The firmware supports a range of Flash sizes, though 4Mbyte seems to be 
+// currently the most common.  Current builds include a discovery function
+// which is enabled by FLASH_AUTOSIZE, but you can override this by commenting
+// this out and enabling the explicitly size, e.g. FLASH_4M.  Valid sizes are
+// FLASH_512K, FLASH_1M, FLASH_2M, FLASH_4M, FLASH_8M, FLASH_16M.
+  
 #define FLASH_AUTOSIZE
+//#define FLASH_4M
+
+
+// The firmware now selects a baudrate of 115,200 by default, but the driver also 
+// includes automatic baud rate detection at start-up by default.  If you want to
+// change the default rate then vaild rates are  300, 600, 1200, 2400, 4800, 9600, 
+// 19200, 31250, 38400, 57600, 74880, 115200, 230400, 256000, 460800 [, 921600, 
+// 1843200, 368640].  Note that the last 3 rates are not recommended as these 
+// might be unreliable.
+ 
+#define BIT_RATE_DEFAULT BIT_RATE_115200
+#define BIT_RATE_AUTOBAUD
+
+
+// Three separate build variants are now supported. The main difference is in the
+// processing of numeric data types.  If LUA_NUMBER_INTEGRAL is defined, then
+// all numeric calculations are done in integer, with divide being an integer 
+// operations, and decimal fraction constants are illegal.  Otherwise all
+// numeric operations use floating point, though they are exact for integer
+// expressions < 2^53.  The main advantage of INTEGRAL builds is that the basic
+// internal storage unit, the TValue, is 8 bytes long, rather than the default 
+// on floating point builds of 16 bytes.  We have now also introduced an 
+// experimental option LUA_PACK_TVALUES which reduces the floating point TValues
+// to 12 bytes without any performance impact.
+
+//#define LUA_NUMBER_INTEGRAL
+//#define LUA_PACK_TVALUES
+
+
+// The Lua Flash Store (LFS) allows you to store Lua code in Flash memory and
+// the Lua VMS will execute this code directly from flash without needing any 
+// RAM overhead.  If you want to enable LFS then set the following define to
+// the size of the store that you need.  This can be any multiple of 4kB up to 
+// a maximum 256Kb.
+
+//#define LUA_FLASH_STORE 0x10000
+
+
+// By default Lua executes the file init.lua at start up.  The following
+// define allows you to replace this with an alternative startup.  Warning:
+// you must protect this execution otherwise you will enter a panic loop.
+// The example provided executes the LFS module "_init" at startup or fails
+// through to the interactive prompt.
+// ********* WARNING THIS OPTION ISN'T CURRENTLY WORKING
```

Comment:
:(

Reply:
I do mean to get this working soon, but it was just a Q of priorities.  It's not absolutely essential for the first merge.